en/components/hierarchicalgrid/load-on-demand.md

```diff
@@ -8,292 +8,206 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
 
 ````TypeScript
 public ngAfterViewInit() {
-    this.remoteService.getData({ parentID: null, rootLevel: true, key: "Customers" }).subscribe((data) => {
-        this.hGrid.data = data;
-        this.hGrid.cdr.detectChanges();
-    });
-}
-````
-
-Next, we only need to create our `gridCreated` method that will request data for any new child grid created. It will be similar to getting the root level grid data, just this time we will need to pass more information, like `parentID` and `parentKey`. `rootLevel` will be `false` for any child:
-
-````TypeScript
-public gridCreated(event: IGridCreatedEventArgs, _parentKey: string) {
-    const dataState = {
-        key: event.owner.key,
-        parentID: event.parentID,
-        parentKey: _parentKey,
-        rootLevel: false
-    };
-    this.remoteService.getData(dataState).subscribe(
-        (data) => {
-            event.grid.data = data;
-            event.grid.cdr.detectChanges();
-        }
-    );
+    this.refreshRootGridData();
 }
 ````
 
-With this, the setup of our application is almost done. This last step aims to improve the user experience by informing the user that the data is still loading so he doesn't have to look at an empty grid in the meantime. That's why the [`IgxHierarchicalGrid`]({environment:angularApiUrl}/classes/igxhierarchicalgridcomponent.html) supports a loading indicator that can be displayed while the grid is empty. If new data is received, the loading indicator will hide and the data will be rendered. 
+#### Getting child grids data
 
-#### Setup of loading indication
+Next we will define how we should build our URL for the GET request in order to get the data for our child grids. This is visual representation of the relations between the tables:
 
-The [`IgxHierarchicalGrid`]({environment:angularApiUrl}/classes/igxhierarchicalgridcomponent.html) can display a loading indicator by setting the [`isLoading`]({environment:angularApiUrl}/classes/igxhierarchicalgridcomponent.html#isloading) property to `true` while there is no data. We need to set it initially for the root grid and also when creating new child grids, until their data is loaded. We could always set it to `true` in our template, but we want to hide it and display that the grid has no data if the service returns an empty array by setting it to `false`.
+<img class="responsive-img" src="../../images/hgrid-database.jpg" />
 
-In this case the final version of our `hierarchical-grid-lod.component.ts` would look like this:
+Finally, we need to create our `gridCreated` method that will request data for any new child grid created. It will be similar to getting the root level grid data, just this time we will use the data provided in the event [`gridCreated`]({environment:angularApiUrl}/classes/igxrowislandcomponent.html#gridCreated) and build our URL with it:
```

Comment:
```suggestion
Finally, we need to imlement our `gridCreated` method that will request data for any new child grid created. It will be similar to getting the root level grid data, just this time we will use the data provided in the event [`gridCreated`]({environment:angularApiUrl}/classes/igxrowislandcomponent.html#gridCreated) and build our URL with it:
```